src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -23,6 +24,7 @@ export class DatasetService {
   private indexPatterns?: IndexPatternsContract;
   private defaultDataset?: Dataset;
   private typesRegistry: Map<string, DatasetTypeConfig> = new Map();
+  private recentDatasets: LRUCache<string, Dataset>;
```

Comment:
is this just in memory or stored in some storage?

Reply:
Currently, we'll just store it in memory.